### DISCUSSION TITLE
Example where the script matters

### DISCUSSION THREAD

**Discussion by wipfli** [harfbuzz/harfbuzz#4426] [Q&A] [answered]
Hi everyone. I'm writing a document about text shaping for people in the maps community.

For that I try to make some examples why the harfbuzz inputs language, direction, and script matter.

While I can find examples when language and direction matter, I have no idea about the script.

Can you maybe help me with an example where the same input unicode text buffer results in different positioned glyphs for different scripts?

Thanks in advance!

**Comment by behdad** [comment, marked as answer]
> Can you maybe help me with an example where the same input unicode text buffer results in different positioned glyphs for different scripts?

The script part is a matter of correctness. With vast majority of fonts in circulation, pretty much for any Arabic or Indic script text, if you set script incorrectly you get wrong results.

**Reply by wipfli** [reply]
Oh that's good to know. Makes sense. Thank you for the quick reply :)

**Reply by khaledhosny** [reply]
If you need a visual evidence:
```
$ hb-view NotoNaskhArabic.ttf كتب --script=arab
```
![a](https://github.com/harfbuzz/harfbuzz/assets/93914/e1b76809-322c-4044-b005-4cfa1e7eb10d)

```
$ hb-view fNotoNaskhArabic.ttf كتب --script=latn
```
![l](https://github.com/harfbuzz/harfbuzz/assets/93914/5986a7b1-3fa0-4f00-9c17-c20fd12b8fd3)

Also note that script should be auto-detected from Unicode character properties, and should not be something the end user be concerned about (just like direction is handled by the bidi algorithm).

**Reply by wipfli** [reply]
Thank you @khaledhosny!

**Reply by wipfli** [reply]
Included the books example here: https://github.com/wipfli/about-text-rendering-in-maplibre#example-6-script-matters
